### PR TITLE
add EdgeTree support for lines/multilines

### PR DESCRIPTION
**Pull request by talevy**
The main change here is that edge-trees originally
checked whether the queried extent could be contained
within its shape. Since line-strings have no inner boundaries,
this check is not useful, the line crosses check + extent-check-bounds
is sufficient.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/geo/EdgeTreeReader.java [hunk 1]
```diff
@@ -33,10 +33,12 @@
 public class EdgeTreeReader implements ShapeTreeReader {
     final ByteBufferStreamInput input;
     final int startPosition;
+    private final boolean closed;
 
-    public EdgeTreeReader(ByteBufferStreamInput input) throws IOException {
+    public EdgeTreeReader(ByteBufferStreamInput input, boolean closed) throws IOException {
```
Comment (imotov) [inline server/src/main/java/org/elasticsearch/common/geo/EdgeTreeReader.java]: Could you add a comment about meaning of closed here?

**Review by imotov — COMMENTED** [review]
Looks good to me. The name `closed` confused me a bit, since we are using it in the context of shapes for a slightly different purpose. I think in this context we want to indicate dimension. `closed == true` means we are dealing with area or 2 dimensional shapes and  `closed == false` means that we are dealing with 1 dimensional shapes.

File: server/src/main/java/org/elasticsearch/common/geo/EdgeTreeReader.java (hunk 1 above)
Comment (talevy) [reply server/src/main/java/org/elasticsearch/common/geo/EdgeTreeReader.java]: will do! I'll update to clarify

A edge-tree representing a shape is open when the edges do not all connect at the beginning and end. it is closed when the edges do connect at the beginning and the end.

**Comment by talevy** [comment]
test this please

**Review by imotov — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**